stately/src/otherNativeMain/kotlin/co/touchlab/stately/concurrency/Lock.kt

```diff
@@ -1,9 +1,21 @@
 package co.touchlab.stately.concurrency
 
+import co.touchlab.stately.freeze
 import kotlin.native.concurrent.AtomicInt
+import kotlin.native.concurrent.ThreadLocal
 
+@ThreadLocal
+private object Thread {
+  val id: Int by lazy { Any().hashCode() }
+}
+
+/*
+ * This is a re-entrant implementation of Lock.
```

Comment:
TBH, this probably belongs on the `expect class`

Reply:
Fair point. I think it would probably be best to rename the class to `ReentrantLock` to make it obvious to anyone using the class. For now i'll add a comment to the expect class though.

Reply:
I agree, but it would be a breaking change.

Reply:
This relies on `Any().hashCode()` to be unique? Looking at the native impl, it'll use the object pointer val, which will be unique unless you get a recycled memory block. Unlikely but no impossible. Why not use a global AtomicInt and increment?